addons/l10n_in/models/res_partner.py

```diff
@@ -85,6 +86,31 @@ def _onchange_l10n_in_gst_status(self):
                 partner.l10n_in_gstin_verified_status = False
                 partner.l10n_in_gstin_verified_date = False
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        partners = super().create(vals_list)
+        if 'import_file' in self.env.context:
+            return partners
+        for partner in partners:
+            if partner.vat and self.check_vat_in(partner.vat):
+                partner.l10n_in_pan_entity_id = partner._l10n_in_get_pan_entity_from_vat(partner.vat).id
+        return partners
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'import_file' in self.env.context:
+            return res
+        if vals.get('vat') and self.check_vat_in(vals['vat']):
```

Comment:
but maybe here add 'or vals.get('country_id', self.env['res.country']).code == 'IN'
...